core/src/ProblemManager.cc

```diff
@@ -354,25 +431,25 @@ void ProblemManager::updateParticleGradients(
         // Scale the velocity gradient.
         for ( int i = 0; i < space_dim; ++i )
             for ( int j = 0; j < space_dim; ++j )
-                work[i][j] = p.grad_v[i][j] * delta_t;
+                work_local[th][i][j] = p.grad_v[i][j] * delta_t;
 
         // Compute the deformation gradient increment.
         for ( int i = 0; i < space_dim; ++i )
             for ( int j = 0; j < space_dim; ++j )
                 for ( int k = 0; k < space_dim; ++k )
-                    delta_F[i][j] += work[i][k] * p.F[k][j];
+                    delta_F_local[th][i][j] += work_local[th][i][k] * p.F[k][j];
 
         // Update the deformation gradient.
         for ( int i = 0; i < space_dim; ++i )
             for ( int j = 0; j < space_dim; ++j )
-                p.F[i][j] += delta_F[i][j];
+                p.F[i][j] += delta_F_local[th][i][j];
 
         // Add the scaled velocity gradient to the identity.
         for ( int i = 0; i < space_dim; ++i )
-            work[i][i] += 1.0;
```

Comment:
I think `work` can remain a thread local variable - you should just be able to allocate it inside the loop

Reply:
I made this change, thanks